var/spack/repos/builtin/packages/python/package.py

```diff
@@ -324,6 +331,132 @@ def filter_compilers(self, prefix):
     # ========================================================================
 
     @property
+    def executable(self):
+        """Returns the path to the Python executable, which may vary depending
+        on the version of Python and how it was installed.
+
+        In general, Python 2 comes with ``python`` and ``python2`` executables,
+        while Python 3 only comes with a ``python3`` executable.
+
+        :returns: The full path to the Python executable
+        :rtype: str
+        """
+        # We need to be careful here. If the user is using an externally
+        # installed python, all 3 executables could be in the same directory.
+
+        # Search for `python2` iff using Python 2
+        if (self.spec.satisfies('@:2') and
+                os.path.exists(os.path.join(self.prefix.bin, 'python2'))):
+            exe = 'python2'
+        # Search for `python3` iff using Python 3
+        elif (self.spec.satisfies('@3:') and
+                os.path.exists(os.path.join(self.prefix.bin, 'python3'))):
+            exe = 'python3'
+        # If neither were found, try `python`
+        elif os.path.exists(os.path.join(self.prefix.bin, 'python')):
+            exe = 'python'
+        else:
+            msg = 'Cannot locate python executable in {0}'
+            raise RuntimeError(msg.format(self.prefix.bin))
+
+        # The python executable may be a symlink if it was installed
+        # with Homebrew. Since some packages try to determine the
+        # location of libraries and headers based on the path,
+        # return the realpath
+        return os.path.realpath(os.path.join(self.prefix.bin, exe))
+
+    def python(self):
+        """Returns the Python executable."""
+
+        return Executable(self.executable)
+
+    def print_string(self, string):
+        """Returns the appropriate print string depending on the
+        version of Python.
+
+        Examples:
+
+        * Python 2
+
+          >>> self.print_string('sys.prefix')
+          'print sys.prefix'
+
+        * Python 3
+
+          >>> self.print_string('sys.prefix')
+          'print(sys.prefix)'
+        """
+        if self.spec.satisfies('@:2'):
+            return 'print {0}'.format(string)
+        else:
+            return 'print({0})'.format(string)
+
+    def get_config_var(self, key):
+        """Returns the value of a single variable. Wrapper around
+        ``distutils.sysconfig.get_config_var()``."""
+
+        cmd = 'from distutils.sysconfig import get_config_var; '
+        cmd += self.print_string("get_config_var('{0}')".format(key))
+
+        return self.python('-c', cmd, output=str)
```

Comment:
I'm getting an error here about 4 rather than 1 argument.
It would seem that the line is missing a pair of brackets. But since it's used elsewhere, it might be something less benign, due to the fact that I'm trying to install `boost +python` with python an external package?